File: Work/jsReverse/某钢股份/baogang.py
# -*- coding: utf-8 -*-
"""
@File    : baogang.py
@Author  : Elliot Lin
@Date    : 2026/7/13 23:46
@Project : ReverseNotes
@Github  : https://github.com/ActingLin/ReverseNotes
@Desc    : 
"""
import requests
import json
import execjs

ctx = execjs.compile(open("baogang.js", encoding='utf-8').read())


for pageNum in range(1, 5):
    x_headers_info = ctx.call("x_headers_info", pageNum)
    print(x_headers_info)

    headers = {
        "Accept": "application/json, text/plain, */*",
        "Accept-Language": "zh-CN,zh;q=0.9",
        "Cache-Control": "no-cache",
        "Connection": "keep-alive",
        "Content-Md5": x_headers_info["Content-Md5"],
        "Content-Type": "application/json;charset=UTF-8",
        "Origin": "https://www.baosteel.com",
        "Pragma": "no-cache",
        "Referer": "https://www.baosteel.com/",
        "Sec-Fetch-Dest": "empty",
        "Sec-Fetch-Mode": "cors",
        "Sec-Fetch-Site": "cross-site",
        "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.0.0 Safari/537.36",
        "sec-ch-ua": "\"Not)A;Brand\";v=\"8\", \"Chromium\";v=\"138\", \"Google Chrome\";v=\"138\"",
        "sec-ch-ua-mobile": "?0",
        "sec-ch-ua-platform": "\"Windows\"",
        "x-date": str(x_headers_info["x-date"]),
        "x-nonce": x_headers_info["x-nonce"],
        "x-signature": x_headers_info["x-signature"],
        "x-user": x_headers_info["x-user"]
    }
    url = "https://cmsauth.baowugroup.com/v1/web/api/content/list"
    params = {
        "domainId": "12"
    }
    data = {
        "pageNo": pageNum,
        "pageSize": 12,
        "condition": {
            "nodeId": 436
        }
    }
    data = json.dumps(data, separators=(',', ':'))
    response = requests.post(url, headers=headers, params=params, data=data)

    print(response.json())
    print('-' * 50)

File: Work/jsReverse/11-某钢股份/baogang.js
var CryptoJs = require('crypto-js')

function l(e) {
    for (var n = [], t = Array.of("0", "1", "2", "3", "4", "5", "6", "7", "8", "9"), a = 0; a < e; a++) {
        var c = Math.floor(10 * Math.random());
        n[a] = t[c]
    }
    return n.join("")
}

function s() {
    return parseInt((new Date).getTime() / 1e3)
}

var n = l(10)
var t = s();
var o = "0/56f5cff3cad14853a44782c2c689ab2a";
var i = "13ade1de1eff43ffb821735f352a4148";


headers = {}
headers["x-user"] = o;
headers["x-nonce"] = n;
headers["x-date"] = t;

function x_headers_info(pageNo){
    data = {
        "pageNo": pageNo,
        "pageSize": 12,
        "condition": {
            "nodeId": 436
        }
    }
    var a = CryptoJs.MD5(JSON.stringify(data)).toString()

    headers["Content-Md5"] = a;
    var c = "".concat("POST", "\n").concat("/v1/web/api/content/list?domainId=12", "\nx-user:").concat(o, "\nx-nonce:").concat(n, "\nx-date:").concat(t, "\nContent-Md5:").concat(a, "\n");
    headers["x-signature"] = CryptoJs.HmacSHA1(c, i).toString().toUpperCase();
    return headers;
}

// console.log(x_headers_info(3))